### PR TITLE
Development

### PR DESCRIPTION
**Pull request by nuernbergerc**
# Pull Request

## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. 
If applicable, list (package/module) dependencies that are required for this change.

- Adds primary-coding functions for POP and HAP
- Adds missing platform independent primary-coding-functions
- Adds WHO-Scale and EQ-5D-5L functions to primary-coding-suep.R
- Solve issue #1 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. 
Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Tested locally within napkon-ecu project for all platforms
- [x] Tested locally with test-data for all platforms

**Test Configuration**:
* Tested OS and OS version:

Windows 10 x64
R Version 4.2.2 (2022-10-31)

## Checklist:

- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new code warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules/packages

### CODE REVIEW

File: R/primary-coding-hap.R
[the remark this answers is not in the document]
Comment (mrainers) [reply R/primary-coding-hap.R:1]: I found the following differences:

1. When osfci_0021 = 0, we count this as mild phase, which is a mistake on our behalf. We need to correct this in our script.
2. We don't distinguish between the reason of hospitalisation. So we just end up with the categories mild, moderate and severe. 

Apart from that both computations should end in the same results. I haven't tested it with the data though.

[the remark this answers is not in the document]
Comment (mrainers) [reply R/primary-coding-hap.R:1]: Yes, I think so. 
I changed our code today to correct the calculation for Patients with osfci_0021 = 0.

**Review by miljukov-o — APPROVED** [review]
I haven't checked everything in detail. But I assume the testing went smoothly. So it should work. We just have to keep an eye open for bugs.

**Comment by miljukov-o** [comment]
@nuernbergerc please update your epicodr installation and test on current export data for hap (31.03.2023) and execute primary_coding_hap() and core_analyses_hap() within in napkon-ecu without sourcing the primary coding scripts. The primary coding from epicodr should work, but it doesn't in my case.

**Comment by miljukov-o** [comment]
It makes sense for simplier analyses projects to remove the underscore from the table names, but it obviously leads to side effects in more complex projects. In napkon-ecu we use the read function of secuTrialR and we also use meta data heavily, where the table names also occur with underscore in many places.

So removing the underscore from the table names in napkon-ecu is not an option. But the primary coding world has to be moved out from napkon-ecu someday and not kept or maintained in both. Therefore we need a smart solution.

**Comment by mrainers** [comment]
Yepp, I agree. I've removed the underscore for conveniance reasons. Because I thought it would make the data more accessable if you can write `data$table` instead of ``data$`_table` ``, also because the error messages doesn't always make it clear why `data$_table` does not work. But if the side effects are to problematic we need a another solution.

**Comment by nuernbergerc** [comment]
i will try it out
if it works i will create a new pull request

**Comment by nuernbergerc** [comment]
works with grep instead of grepl
maybe we should consider implementing this solution for all platforms to avoid similar errors if there are changes in the secuTrial exports

**Pull request merged**